sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AlreadyExistException.scala

```diff
@@ -32,19 +33,53 @@ import org.apache.spark.sql.types.StructType
 class DatabaseAlreadyExistsException(db: String)
   extends NamespaceAlreadyExistsException(Array(db))
 
+// any changes to this class should be backward compatible as it may be used by external connectors
+class NamespaceAlreadyExistsException private(
```

Comment:
I tried to support all 3.3 constructors. It is actually tricky as it was a case class before. Therefore, there is no way to make it 100% compatible. I can add a custom companion object to make it act like the old case class.

@gengliangwang, what do you think?

Reply:
>  It is actually tricky as it was a case class before

I took a look at https://github.com/apache/spark/pull/37887/files and `NamespaceAlreadyExistsException` doesn't seem to be a case class.
Anyway, I don't strong opinion. It's up to you.

Reply:
Wrong place to comment. I meant [this](https://github.com/apache/spark/blob/b40e408ea13d9d2d55d407f0758940cbba5ade3e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/NoSuchItemException.scala#L35).